apps/src/aiTutor/views/teacherDashboard/InteractionsTable.tsx

```diff
@@ -58,31 +94,98 @@ const InteractionsTable: React.FC<InteractionsTableProps> = ({sectionId}) => {
   const [selectedTimeFilter, setSelectedTimeFilter] = useState<string>(
     TimeFilter.AllTime
   );
+  const [isLoading, setIsLoading] = useState<boolean>(true);
 
   useEffect(() => {
     (async () => {
       try {
-        const chatMessages = await fetchAITutorInteractions({sectionId});
-        setChatMessages(chatMessages);
+        const messages = await fetchAITutorInteractions({sectionId});
+        setChatMessages(messages);
+        setStudentFilterOptions(generateStudentFilterOptions(messages));
       } catch (error) {
-        console.log('error', error);
+        setChatMessages([]);
       }
+      setIsLoading(false);
     })();
   }, [sectionId]);
 
-  useEffect(() => {
-    const uniqueStudentFilterOptions = chatMessages.reduce<CheckboxOption[]>(
-      (acc, message) => {
-        const userId = `${message.userId}`;
-        if (!acc.some(student => student.value === userId)) {
-          acc.push({label: message.studentName, value: userId});
-        }
-        return acc;
+  const generateStudentFilterOptions = (messages: StudentChatRow[]) => {
```

Comment:
Extracting this into a generic "generateOptions" helper allowed me to combine the two `useEffect` blocks into one.